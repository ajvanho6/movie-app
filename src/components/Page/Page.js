import React from 'react';
import PropTypes from 'prop-types';

import './Page.scss';

const Page = ({
    footer,
    content,
    navigation,
}) => {
    return (
        <div className="m-app-page">
            {navigation || null}
            <div className="m-app-page__content">
                {content}
            </div>
            {footer}
        </div>
    );
};

Page.propTypes = {
    navigation: PropTypes.element,
    footer: PropTypes.element,
    content: PropTypes.element,
};

Page.defaultProps = {
    navigation: null,
    footer: null,
    content: null,
};

export default Page;
