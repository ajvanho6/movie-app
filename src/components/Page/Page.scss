@import '../../styles/config';
@import '../../styles/mixins';

$component-class: m-app-page;

.#{$component-class} {
    position: relative;
}
