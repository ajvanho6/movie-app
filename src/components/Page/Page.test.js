import React from 'react';
import {shallow} from 'enzyme';
import Page from './Page';

describe('Page', () => {
    it('Page should render correctly', () => {
        const component = shallow(<Page />);
        expect(component).toMatchSnapshot();
    });
});
