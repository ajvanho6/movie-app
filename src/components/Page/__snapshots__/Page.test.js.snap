// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Page should render correctly 1`] = `ShallowWrapper {}`;
